Shared/Notifications/NotificationsManager.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  NotificationManager.swift
+//  Eatery
+//
+//  Created by Ethan Fine on 2/17/20.
+//  Copyright © 2020 CUAppDev. All rights reserved.
+//
+
+import SwiftyUserDefaults
+import UserNotifications
+
+class NotificationsManager {
+    
+    static let shared = NotificationsManager()
+    
+    private let favoriteMenuItemNotifTitle = "Grub on one of your favorite foods!"
+    private let notifContentDateKey = "date"
+    private let notifContentEateriesKey = "eateryNames"
+    private let notifContentEventNameKey = "eventName"
+    private let notifContentMenuItemKey = "menuItemName"
+    /// Maps menu item names to an an array of scheduled notification requests
+    private var scheduledNotifications = [String : [UNNotificationRequest]]()
+    
+    /// Requests authorization to the user for notifications
+    func requestAuthorization() {
+        var authorizationOptions: UNAuthorizationOptions = [.alert]
+        if #available(iOS 13.0, *) {
+            authorizationOptions.insert(.announcement)
+        }
+        UNUserNotificationCenter.current().requestAuthorization(options: authorizationOptions) { (granted, error) in }
+    }
+    
+    /// Map scheduled notifications to the item they are scheduled to notify about
+    private func populateScheduledNotifications() {
+        UNUserNotificationCenter.current().getPendingNotificationRequests(completionHandler: { (requests) in
+            for request in requests {
+                let notifInfo = request.content.userInfo
+                guard let menuItemName = notifInfo[self.notifContentMenuItemKey] as? String else { continue }
+                if var scheduledNotifications = self.scheduledNotifications[menuItemName] {
```

Comment:
do you need the `self` in `self.scheduledNotifications`?

Reply:
Yep, required